### PR TITLE
Extend the reference team to the lang-docs team

### LINKED ISSUE #443: Set up for the t-lang/doc team
I talked with Josh and other folks, and propose to extend the current reference team to the "lang docs" team in order to maintain more docs like nomicon.

For context: https://rust-lang.zulipchat.com/#narrow/stream/196563-t-lang.2Fwg-meta and https://github.com/rust-lang/lang-team/issues/47
cc @rust-lang/lang for the approval

### PR DESCRIPTION
**Pull request by JohnTitor**
As mentioned in #443, this extends the reference team to the lang-docs team.
This is the first step, we could add folks interested in maintaining the docs later.
Added myself to the leads as a maintainer of the nomicon.

Closes #443

cc @rust-lang/lang, and @ehuss as they maintain the docs submodules on rust-lang/rust.

### CODE REVIEW

**Comment by pietroalbini** [comment]
Note that deploying this will need someone with admin permissions to first rename the team on the GitHub interface, otherwise we'll lose all permissions configured in the repositories: `sync-team` doesn't handle renames, it just removes and then recreates the team.

**Comment by JohnTitor** [comment]
@pietroalbini If my understanding is correct, the current "reference" team doesn't use `sync-team` as it doesn't have the `[[github]] orgs = ["rust-lang"]` section, right?

**Comment by JohnTitor** [comment]
But yeah, we could archive the reference team then create a fresh team if it's desirable.

**Comment by pietroalbini** [comment]
@JohnTitor I think the best solution is for someone to rename the old team to the new one before merging this PR, and then merge this as-is once someone from the lang team approves the changes.

**Comment by JohnTitor** [comment]
@pietroalbini 
> for someone to rename the old team to the new one before merging this PR

You mean the team is on GitHub? We don't have the reference team on the GitHub teams AFAIK.

**Comment by pietroalbini** [comment]
Sorry, you're right :)

Then this PR is fine to merge as soon as someone from lang acks.

File: teams/lang-docs.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name = "lang-docs"
+subteam-of = "lang"
+
+[people]
+leads = ["ehuss", "JohnTitor"]
+members = [
+    "ehuss",
+    "Havvy",
+    "JohnTitor",
```
Comment (Manishearth) [inline teams/lang-docs.toml:9]: @ehuss @havvy @matthewjasper Wanted to make sure y'all were okay with this; I do not see many PRs from @JohnTitor on the nomicon and zero reviews.

It's atypical to add new maintainers in such scenarios, typically team members get added once they've had some involvement already.

Of course, if @JohnTitor has been involved in other, less visible ways, and y'all think they'd be a good addition to the team, ignore me.

**Comment by JohnTitor** [comment]
On #443, Josh and Taylor from the lang added 👍 so we could say this got some approvals?

The core team (i.e. org owner) only can merge this and set up for repository permission so could I ask @nikomatsakis as a core and lead of the lang to do that?

**Comment by nikomatsakis** [comment]
Looks great!

File: teams/lang-docs.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "Centril",
+]
+
+[[github]]
+orgs = ["rust-lang"]
+
+[website]
+name = "lang-docs team"
+description = "Developing and writing the docs related to the lang team"
+zulip-stream = "t-lang/docs"
```
Comment (ehuss) [inline teams/lang-docs.toml:22]: This link is broken.  The zulip stream is `t-lang/doc`.

Comment (JohnTitor) [reply teams/lang-docs.toml:22]: Oops, thanks. Will fix it soon.

**Pull request merged**